netkan/netkan/status.py

```diff
@@ -36,6 +36,12 @@ class Meta:
     last_inflated = UTCDateTimeAttribute(null=True)
     success = BooleanAttribute()
 
+    homepage = UnicodeAttribute(null=True)
+    spacedock = UnicodeAttribute(null=True)
+    repository = UnicodeAttribute(null=True)
+    curse = UnicodeAttribute(null=True)
+    bugtracker = UnicodeAttribute(null=True)
+
```

Comment:
Adding a bunch of specifically named fields isn't ideal, we either want to store a JSON payload (each field can hold up to 400Kb) or have a generic + type. I'm leaning towards a `resources` field, means we don't need to add a PR everytime we want to add some data (we're not doing any sort of filtering on this here). Though we should be aware that if our export grows beyond 10,240Kb we'll need to think about what we need to do (we're at like 400Kb, so we shouldn't have the problem for a while and I'd hope we've got everything polished enough to build an API).

Reply:
I'm very open to changing this, but I don't quite follow what you're suggesting. What would a generic + type look like?

Reply:
We can use a [map type](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/HowItWorks.NamingRulesDataTypes.html#HowItWorks.DataTypes). If we wanted to, we could just put the resources directly in there as they appear in the metadata. Which in future if we wanted to map this data in an API directly, we could retrieve it from different places. I'm actually leaning towards this option, because it matches the metadata as is (even though from a data structure layout it doesn't scale well).

In regards to mapping, I mean defining types (don't take these names as gospel)

```json
{
    "resources": {
        "provider": "github",
        "provider_uri": "https://github.com/foo/bar",
        "homepage": "https://foo.com/bar"
    }
}
```

Means that adding a provider doesn't involve spec changes, reduces code changes etc etc. But as noted, I think a resources map datatype and just put the relevant fields in there. If we want to change how we do resources in the future, doing it at the indexing end doesn't make sense. We're consuming the spec, not deciding it.

Reply:
OK, replaced the discrete properties with one `resources` `MapAttribute`.